### ISSUE TITLE
[Feature Request] Undo history when editing cell

### ISSUE THREAD

**Issue by JorgeSivil** [dbeaver/dbeaver#1446]
Basic undo when editing cell (Control + Z) when inside "Editing Cell" (Enter key)

Example:

Current Value: "Something"
Edited Value: "Something more"
Press Control + Z, then value becomes "Something" again.

**Comment by serge-rider** [comment]
You can revert value changes by pressing ESC on a cell.

**Comment by JorgeSivil** [comment]
No no, I'm not taking about "Cancel complete change", I'm talking about undoing string changes.

For the case I put, it was trivial and doable with Escape.

But when you edited the string 6 times, it would be useful. Mostly for 'text' fields I think.

Example:

This text is very very large and I would like to edit it. (Persisted Value)
This text is very very large and I would like to edit it to add more things because I really like to edit. (edit in progress)
This text is very very large and I would like to edit it to add more text. (edit in progress)
This text is very very large and I would like to edit it to add more things because I really like to edit. (control + z returned to the first edit)

Just like a text editor.

**Comment by allanlaal** [comment]
I would also request that the larger edit dialog (Shift+Enter on a cell) would have undo history

**Comment by serge-rider** [comment]
@allanlaal did you try to edit value in value panel? Undo/redo work there.

**Comment by allanlaal** [comment]
@serge-rider no, I meant the large multiline extended content editor referenced in #2634 that opens up with Shift+Enter
should this request be a separate ticket?

**Comment by kseniiaguzeeva** [comment]
JorgeSivil, please reopen the ticket if it is still actual for you.

**Comment by allanlaal** [comment]
its actual for me still :)

**Comment by allanlaal** [comment]
@kseniiaguzeeva it's Actual for me still :)

**Comment by ghost** [comment]
@kseniiaguzeeva It's Actual for me too. Thank you.

**Comment by lokesh1729** [comment]
is this issue triaged? any update on this?

**Comment by eliliam** [comment]
Is this something that will ever be built? Currently the only way to clear/undo uncommitted/unsaved changes in the data viewer is to close the table tab and discard unsaved changes. This works, but in practice it is very inconvenient. Having the ability to undo any pending/uncommitted changes in the data viewer is a crucial feature DBeaver currently lacks. Could we please get an update on this, even if it is to say that it will not be addressed?
